ui/src/app/pages/login/sign-in.tsx

```diff
@@ -71,99 +60,188 @@ const styles = {
+          return <AccountCreation invitationKey={this.state.invitationKey}
+                                  profile={this.state.profile}
+                                  onComplete={(profile: Profile) => this.setState({
+                                    profile: profile,
+                                    // Skip over the demographic survey if new-style form isn't enabled.
+                                    currentStep: enableNewAccountCreation ? SignInStep.DEMOGRAPHIC_SURVEY :
+                                      SignInStep.SUCCESS_PAGE
+                                  })}/>;
+        case SignInStep.DEMOGRAPHIC_SURVEY:
+          return <AccountCreationSurvey
+            profile={this.state.profile}
+            invitationKey={this.state.invitationKey}
+            termsOfServiceVersion={this.state.termsOfServiceVersion}
+            onComplete={(profile: Profile) => this.setState({
+              profile: profile,
+              currentStep: SignInStep.SUCCESS_PAGE
+            })}
+            onPreviousClick={(profile: Profile) => this.setState({
+              profile: profile,
+              currentStep: SignInStep.ACCOUNT_CREATION
+            })}/>;
+        case SignInStep.SUCCESS_PAGE:
           return <AccountCreationSuccess profile={this.state.profile}/>;
         default:
           return;
       }
     }
 
-    setCurrentStep(nextStep: Step) {
-      this.setState({
-        currentStep: nextStep
-      });
-    }
-
-    onKeyVerified(invitationKey: string) {
-      this.setState({
-        invitationKey: invitationKey,
-        currentStep: {stepName: 'accountCreation'}
-      });
-    }
-
-    setProfile(profile, currentStep) {
-      this.setState({
-        profile: profile,
-        currentStep: currentStep
-      });
-    }
-
-
-
     render() {
-      const {stepName, backgroundImages} = this.state.currentStep;
-      const maxWidth = backgroundImages === undefined ? '100%' : '41.66667%';
-      return <div style={styles.signedInContainer}>
-        <FlexColumn style={{width: '100%'}}>
-          <div data-test-id='template'
-               style={styles.template(this.props.windowSize, backgroundImages)}>
-            <img style={{height: '1.75rem', marginLeft: '1rem', marginTop: '1rem'}}
-                 src={headerImg}/>
-            <div style={{flex: `0 0 ${maxWidth}`,
-              maxWidth: maxWidth, minWidth: '25rem'}}>
-              {this.nextDirective(stepName)}
-            </div>
-          </div>
+      const backgroundImages = StepToImageConfig[this.state.currentStep];
+      return <FlexColumn style={styles.signInContainer} data-test-id='sign-in-container'>
+        <FlexColumn data-test-id='sign-in-page'
+             style={styles.template(this.props.windowSize, backgroundImages)}>
+          <div><img style={{height: '1.75rem', marginLeft: '1rem', marginTop: '1rem'}}
+                    src={HEADER_IMAGE}/></div>
+          {this.renderSignInStep(this.state.currentStep)}
```

Comment:
One thing that looks a bit wrong is the lack of any margin under the "submit" button on the demographics survey. That might be an existing issue, so feel free to ignore it here. Note, this screenshot shows the bottom of the window, fully scrolled:

![image](https://user-images.githubusercontent.com/822298/73900863-dfa07680-4845-11ea-8bc4-cc9035c4cc7d.png)